src/components/Composer/index.js

```diff
@@ -306,7 +306,11 @@ class Composer extends React.Component {
      * @param {ClipboardEvent} event
      */
     handlePaste(event) {
-        if (!this.props.checkComposerVisibility() && !this.state.isFocused) {
+        if (!this.props.checkComposerVisibility()) {
+            return;
+        }
+
+        if (['INPUT', 'TEXTAREA'].includes(event.target.nodeName)) {
```

Comment:
@robertKozik This is flawed in the root itself. ` this.props.checkComposerVisibility()` always fails for edit message items as we didn't pass `checkComposerVisibility`

Reply:
This event is called `twice` one from each instance. We should perform the operation only when it's from the `textInput`

Reply:
I came up with a quick solution here
```
if (!this.props.checkComposerVisibility() && !this.textInput.isFocused()) {
            return;
        }

        if (this.textInput.id !== event.target.id) {
            return;
        }
```

Reply:
Yeah you're right. I'm also looking for the another fix for this one. Did you check if pasting to the edited comment works?

Reply:
I'm not sure that paste event should be trigger only from focused input. The "parent" issue and PR https://github.com/Expensify/App/pull/21583 was introducing the ability to paste when input is not composed

Reply:
No, it didn't work.

Reply:
also, `checkComposerVisiblity` would always be true when no function is passed as the default prop always returns false.

Currently I'm experimenting with introducing textInput-scoped paste listener alongside the document-scoped one. When event is triggered from textInput/textArea document-scoped will do nothing, and textInput-scoped will do the hard work

Reply:
Here is the snippet
        ```js
const isComposerVisible = this.props.checkComposerVisibility() 
        const isFocused = this.textInput.isFocused()
        if (!(isComposerVisible || isFocused)) {
            return;
        }

```

Reply:
@robertKozik This works for me. Address me all cases.

cc: @francoisl @mountiny

Reply:
This snippet addresses both issues, but now if I edit an existing message and paste from the clipboard, the cursor remains at the beginning of the pasted message:

https://github.com/Expensify/App/assets/2229301/d6126975-0631-417e-aab6-6c47c80ee72a

I'd say let's revert this for now, and then work on a proper fix without having to rush.

Reply:
Yes agreed lets do that first!

Reply:
Great @Santhosh-Sellavel It works for me too

Reply:
This [one](https://github.com/Expensify/App/pull/22817#discussion_r1263034541) comes out of nowhere.

Please revert this first and work on the other issue!

Reply:
Revert open here https://github.com/Expensify/App/pull/22845, feel free to review.

Reply:
@robertKozik about the cursor [issue](https://github.com/Expensify/App/pull/22817/files#r1263034541)

The contributor [here](https://github.com/Expensify/App/issues/22833#issuecomment-1635174418) found the PR [#22443](https://github.com/Expensify/App/pull/22443) which causes it